### PR TITLE
Added CircleCI configuration linter

### PR DESCRIPTION
**Pull request by docwhat**
This is a simple linter definition for the `circleci` CLI tool.

I removed the `<!-- trunk-ignore(markdownlint/MD041) -->` because trunk told me it wasn't needed.  If that was wrong, I could put it back.

I'm really liking the `trunk` tool and appreciate how well-designed the tool is.

I would suggest adding a `CONTRIBUTING.md` file with instructions for what you would expect for a well-formed linter or action.

### CODE REVIEW

**Comment by sxlijin** [comment]
Hi there! Thanks for using trunk and for this contribution - we're glad to have you on board :) (and that you enjoy the design decisions we've made!)

We're actually currently in the process of setting up a testing framework for linters defined here - right now we have a whole bunch of linter integration tests that live alongside the CLI tests (where we test all the default configs that we ship trunk with), but we're working on pulling a whole bunch of those into plugins. That's actually what all the "Plugins Test Runner" failures are 😅 

We can probably go ahead and get this PR merged without passing those tests, but let me circle back with the team to be sure we're OK with that.

**Comment by docwhat** [comment]
Cool!  Thanks!

**Comment by docwhat** [comment]
Thanks for the help, @sxlijin!

**Comment by sxlijin** [comment]
:)

We're about to merge #68 which should also solve the test failures you're seeing right now.

File: readme.md [hunk 1]
```diff
@@ -1,4 +1,3 @@
-<!-- trunk-ignore(markdownlint/MD041) -->
 <p align="center">
```
Comment (sxlijin) [inline readme.md:1]:
```suggestion
<!-- trunk-ignore(markdownlint/MD041) -->
<p align="center">
```

**Review by sxlijin — APPROVED** [review]

**Pull request merged**